### DISCUSSION TITLE
Give each package a separate color in the terminal

### DISCUSSION THREAD

**Discussion by m11m** [vercel/turborepo#1877] [Ideas]
![image](https://user-images.githubusercontent.com/92771507/188791635-42e25a67-c870-4dc5-aa8c-9ea805d0a42f.png)

Right now each package gets a random color when you run something like `turbo run build`, which means sometimes 2 packages will have the same color (see picture, `docs` and `client` look the same). This is a really minor thing but it would be nice to see each package differentiated better at a glance by making sure there are no duplicates 🙏 ty

**Comment by franky47** [comment]
I would also suggest making the color decision stable.

At the moment, because of the randomness, a turborepo containing a single workspace (not very useful, but you have to start somewhere) gets a (potentially) different color every time a command is executed.

A stable coloring option suggestion would be to hash the package.json name, reduce this information to a given number of bits, and use it as the index of a color table.

In any case, for projects with more workspaces than color options, there will be duplication. The use of a larger color palette (eg: 256 colors, or the full 24bit RGB color space) may solve that, at the expense of compatibility (not all TTYs can render those colors) and accessibility (colors may not be contrasted enough on an unknown terminal background color).

Edit: the color is not random, but is based on the cache hash, with 5 possible color values:
https://github.com/vercel/turbo/blob/7f1bb8a504eb385c491a4835dc2f340e7f07eba1/cli/internal/colorcache/colorcache.go#L34-L52

**Comment by anthonyshew** [comment]
A lot has changed since this Discussion was made but I'll close this in favor of https://github.com/vercel/turbo/issues/2564.

P.S. There's a new terminal UI in 2.0 to try out where this wouldn't be so much of an issue.
